test/jdk/javax/swing/event/FocusEventCauseTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+*/
+
+/*
+ * @test
+ * @key headful
+ * @bug 8306119
+ * @summary Verifies FocusEvent cause when components requests focus
+ *          via mouse events
+ * @run main FocusEventCauseTest
+ */
+
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Robot;;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.awt.event.InputEvent;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+public class FocusEventCauseTest {
+    static JFrame frame;
+    static JButton button1;
+    static JButton button2;
+    static volatile Point pt;
+    static volatile Dimension dim;
+    static volatile boolean expected;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+            SwingUtilities.invokeAndWait(() -> {
+                frame = new JFrame("FocusEventCauseTest");
+                JPanel panel = new JPanel();
+                button1 = new JButton("Button1");
```

Comment:
Generic test comment. Its good if we can check focus event for all UI components that are getting updated.

Reply:
Actually, the fix is trivial, the test is only added to show how the cause whould be presented to the user using a widget.
The other widgets would be same in showing focus cause event to be MOUSE_EVENT so not handled separately..